agent/attacher/src/jp/co/ntt/oss/heapstats/attacher/AgentAttacher.java

```diff
@@ -58,6 +58,23 @@ public static void attachProcess(VirtualMachineDescriptor vmd,
                                         vmd.displayName());
       }
 
+      /* Recommend jcmd instead of Attach API if jdk9 or later */
+      /* See https://bugs.openjdk.java.net/browse/JDK-8177154 */
+      String[] versions = vm.getSystemProperties().getProperty("java.version").split("\\.", 3);
+      /* Check early access. See https://bugs.openjdk.java.net/browse/JDK-8061493 */
```

Comment:
According to [JDK-8061493](https://bugs.openjdk.java.net/browse/JDK-8061493), we might use `java.specification.version` instead of `java.version` because specification version does not include pre-release or not.

If we run HeapStats on JDK 8, this property will be set to `1.8` . If JDK 9, this property will be set to `9` . So we can judge version string more easily.